### ISSUE TITLE
App crashes when `swap` called on SearchDirectionView and one direction is null

### ISSUE THREAD

**Issue by CodeFactoryPDabrowski** [Mapwize/mapwize-ui-android#11]
Error log: 
```java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String io.indoorlocation.core.IndoorLocation.getProvider()' on a null object reference
        at io.mapwize.mapwizeformapbox.map.MapwizeIndoorLocation.<init>(Unknown Source)
        at io.mapwize.mapwizecomponents.ui.SearchDirectionView.setToDirectionPoint(SearchDirectionView.java:269)
        at io.mapwize.mapwizecomponents.ui.SearchDirectionView.swap(SearchDirectionView.java:367)
        at io.mapwize.mapwizecomponents.ui.SearchDirectionView.lambda$initialize$2(SearchDirectionView.java:96)
        at io.mapwize.mapwizecomponents.ui.-$$Lambda$SearchDirectionView$0lAgWSWt-h5W_tpwM3Jq9C8QMss.onClick(lambda)
        at android.view.View.performClick(View.java:5637)
        at android.view.View$PerformClick.run(View.java:22393)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:173)
        at android.app.ActivityThread.main(ActivityThread.java:6459)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:938)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:828) ```

Steps to reproduce:
1. Open Map
2. Tap at any object (pin) on the map to open information bottom bar
3. Tap at "Direction" from the bottom bar 
4. Type something in input fields e.g. "qwerty"
5. Try to swap directions

Video with a crash
https://monosnap.com/file/29gcoD97pB8CGGDSFEir2ZQmucHnkR

**Comment by emerciercontexeo** [comment]
Hi,

Thanks for reporting this issue.

I will try to update it as soon as possible.

**Comment by kontownik** [comment]
@emerciercontexeo any update on this one?

**Comment by emerciercontexeo** [comment]
I just push the 1.0.3.

The swap button is now hidden when the search is open. The button does not make sense during a search.

**Comment by CodeFactoryPDabrowski** [comment]
Hi @emerciercontexeo I've checked v1.0.3 and because of change in this line https://github.com/Mapwize/mapwize-ui-android/pull/13/files#diff-914b46768c98aaddb374bb67cce0bba0R239 it's crashing after the second click on swap button :'( This condition && mapwizePlugin.getUserPosition() != null causes that execution moves to this line https://github.com/Mapwize/mapwize-ui-android/pull/13/files#diff-914b46768c98aaddb374bb67cce0bba0R254 and NPE is throwing:'( Could you remove this additional statement from this if?

**Comment by emerciercontexeo** [comment]
@CodeFactoryPDabrowski 

I released the 1.0.4 with some improvements in the direction search view.

The behavior looks clearly better now.
